File: TPs/tp1/TP01Q02 - Palíndromo em C/palindromo.c
/*
     walker freitas dos santos
     698774
*/

#include <stdio.h>     // para entradas e saida
#include <stddef.h>    // para definicoes basicas
#include <stdlib.h>    // para a biblioteca padrao
#include <string.h>    // para cadeias de caracteres
#include <stdarg.h>    // para tratar argumentos
#include <stdbool.h>   // para definicoes logicas
#include <ctype.h>     // para tipos padroes
#include <math.h>      // para definicoes matematicas
#include <time.h>      // para medir tempo



/**
     Função para verificar se a cadeia de caracteres é igual a fim

     @param: string a ser analisada
     @return: true caso a string seja igual a procurada
               false caso a string nao seja igual a procurada
*/
bool fim (char *palavra)
{
	return (strlen(palavra) == 4 && (palavra[0] == 'F') && palavra[1] == 'I' && palavra[2] == 'M');
}


/**
     Função para verificar se a cadeia de caracteres é um palindromo

     @param: string a ser analisada
     @return: true caso a string seja igual um palindromo
               false caso a string nao seja igual a um palindromo
*/
bool pali (char *palavra)
{
     bool resp = true;

     for (int i = 0, j = (strlen(palavra) - 2); i < j; i++, j--)
     {//dois verificadores que vão comparando cada letra
    
     	if(palavra[i] != palavra[j])
          {//se forem diferentes o codigo ja para e retorna
     		resp = false; 
     		i = j; 
     	}
     }
     return resp;
}


int main ()
{
	char *palavra = (char*) (calloc (1000, sizeof(char)));
     fgets(palavra, 999, stdin);

     while(!fim(palavra))
     {       
     	if(pali(palavra))
          {
     		printf("SIM\n");    
     	}
          else
          {
     		printf("NAO\n");
     	}
          fgets(palavra, 999, stdin);
	}
 
}

File: TPs/tp1/TP01Q01 - Palíndromo em Java/palindromo.java
/*
    Aluno: walker freitas dos santos
    N: 698774
*/

class palindromo{

     /**
          Metodo para verificar um certa palavra usada como ponto de parada
          
          @para: sring para ser analisada;
          @retur: true caso a palavra seja igual a necessitada
                  false caso a palavra não seja igual
     */
	public static boolean fim (String palavra){ //ver se é a palavra chave
		return(palavra.length() >= 3 && palavra.charAt(0) == 'F' && palavra.charAt(1) == 'I' && palavra.charAt(2) == 'M');
	}

     /**
          Metodo para verificar um certa palavra é um palindromo
          
          @para: sring para ser analisada;
          @retur: true caso a palavra seja um palindromo
                  false caso a palavra não seja um palindromo
     */
	public static boolean pali (String palavra){
		boolean resp = true;
	
		for (int i = 0, j = (palavra.length() - 1); i < j; i++, j--){//dois verificadores que vão comparando cada letra
			if(palavra.charAt(i) != palavra.charAt(j)){//se forem diferentes o codigo ja para e retorna
				resp = false; 
				i = j; 
			}
		}
		return resp;
	}

	public static void main(String[] args){
		 	
		String resp = MyIO.readLine();	
		
		while(!(fim(resp))){

			if(pali(resp)){
				System.out.printf("SIM\n");
			}else{
				System.out.printf("NAO\n");
			}
          	resp = MyIO.readLine(); 
     	}
	}	
}
